include/swift/SIL/OwnershipUtils.h

```diff
@@ -520,7 +520,6 @@ class InteriorPointerOperandKind {
   InteriorPointerOperandKind(Kind newValue) : value(newValue) {}
 
   operator Kind() const {
-    assert(isValid());
```

Comment:
The reason why is that this operator is used to check if the underlying kind is invalid or not when performing a bool check.